core/include/kernel/thread.h

```diff
@@ -19,6 +19,7 @@
 #define THREAD_FLAGS_COPY_ARGS_ON_RETURN	BIT(0)
 #define THREAD_FLAGS_FOREIGN_INTR_ENABLE	BIT(1)
 #define THREAD_FLAGS_EXIT_ON_FOREIGN_INTR	BIT(2)
+#define THREAD_FLAGS_SYSTEM_SERVICE		BIT(3)
```

Comment:
What's the point of this?

Reply:
This flags allows to track which active system threads are consumed, in function `enough_free_system_thread()`.